### PR TITLE
Deprecate newaddress p2sh

### PR DESCRIPTION
**Pull request by rustyrussell**
This means we no longer hand out p2sh addresses (after two more releases).  Hopefully we'll be handing out taproot by then!

Based on #5747.

### CODE REVIEW

**Comment by rustyrussell** [comment]
Last DF test fixup due to number change (not using p2sh any more, so we have more sats!) and rebased to fix up rust gen files.

File: tests/test_invoices.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def test_invoice(node_factory, chainparams):
     l1, l2 = node_factory.line_graph(2, fundchannel=False, opts={'log-level': 'io'})
 
     addr1 = l2.rpc.newaddr('bech32')['bech32']
-    addr2 = l2.rpc.newaddr('p2sh-segwit')['p2sh-segwit']
+    addr2 = '2MxqzNANJNAdMjHQq8ZLkwzooxAFiRzXvEz' if not chainparams['elements'] else 'XGx1E2JSTLZLmqYMAo3CGpsco85aS7so33'
```
Comment (vincenzopalazzo) [inline tests/test_invoices.py:18]: we would like to introduce a top constant for the different chains?

I think we should have already something, I remember when I did the change of the default port, but I did not remember where this constant are

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/5751/commits/c530b3dbc866b84131d380f2cfa96fd786b0823a

Rerun the CI because looks like a little bit flacky in the last 2 days (from github side)

File: tests/test_invoices.py (hunk 1 above)
Comment (rustyrussell) [reply tests/test_invoices.py:18]: It's only two places and it's better to be explicit, IMHO.

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/pull/5751/commits/48661a35e22ca6767d45ee4f2b62f71b1b05aed5

Simple fix...

**Pull request merged**